src/python/bcc/__init__.py

```diff
@@ -155,6 +160,63 @@ def generate_auto_includes(cls, program_words):
                         headers += "#include <%s>\n" % header
         return headers
 
+    @classmethod
+    def revise_header_files(self, text):
+        """
+        As for some architectures(especially for arm64), their linux kernel headers files
+        might contain assemble codes and specific register name.
+        However, the clang compiler doesn't recoginize them, and will fail to compile them.
+        Therefore, in order to make the compiler happy, it tries to avoid including them in header files. 
+
+        """
+
+        archtype = os.popen('uname -m').read().strip()
+        if archtype == "aarch64" :
+            extra_header = ""            
+
+            kernel_version=os.popen('uname -r').read().strip()
+            kernel_filelist=[ "/usr/src/kernels/"+kernel_version+"/arch/arm64/include/asm/sysreg.h",
+                              "/usr/src/kernels/"+kernel_version+"/arch/arm64/include/asm/virt.h" ]
+            for filename in kernel_filelist:
+                if os.path.isfile(filename):
+                    filehandle = open(filename)
+       
+                    disable_assemble_flag = True
+                    cur_endif_index = 0
+                    for line in filehandle :
+                        if re.match("#if.*__ASSEMBLY__", line):
```

Comment:
In this example the line 58 is not seen by clang, it's the line 60 that clang complains about,
but I don't understand how this patch removes it. It seems to me that it replaces
#ifdef __ASSEMBLY_
with #if 0,
so the line 60 will still be seen by clang.
What am i missing?
Is the goal to remove all of lines 57-61 ? and not just asm part before #else ?
so for sysreg.h file it will remove the lines 248-327 too ?